jina/types/document/multimodal.py

```diff
@@ -0,0 +1,39 @@
+from typing import List
+
+from . import Document
+from ...excepts import LengthMismatchException, BadDocType
+
+class MultimodalDocument(Document):
+    """Each :class:`MultimodalDocument` should have at least 2 chunks (represent as :class:`DocumentSet`)
+    and len(set(doc.chunks.modality)) == len(doc.chunk)
+    """
+    def __init__(self, document = None,
```

Comment:
Maybe there should be the possibility to build it from N documents with different modalities and to merge them into one multimodal document?

Would that also be useful?

Reply:
Another useful interface would be to extract the embedding or content by modality (or the chunk) given a modality name.

Reply:
We will need that interface, otherwise we do not remove any heavylifting from the user when creating this, no?. Or we will need some kind of MultiModalDocumentBuilder?

We need to decide what experience we want to offer when building a document like this

Reply:
I think we can offer a Builder interface or design this class as a Builder. And delegate the checks of correctness at the last step of the build step

Reply:
I had some thought before, it should be something like an observer. The assurance of the correctness should happens at when we call `chunks.add` or `chunks.append` at `DocumentSet` or `ChunkSet`. But meanwhile I'm afraid it's a bit "over engineering". But it's good to have some discussion over that.

Reply:
For now I would be happy just by adding thr interface to build directly from chunks. So that some boilerplatr can be added. What we would need to know is maybe about the granularities to be assigned and so on

Reply:
the interface add as `from_chunks`. for check of the correctness, now I'm using an internal method called `_validate`, and chunks were validated at 2 places: 1. if we build `MultimodalDocument` using `from_chunks` or constructor, it will be validated inside the constructor. 2. Validate the chunks at `modality_content_mapping` inside `_build_modality_content_mapping`, since `modality_content_mapping` is the common entrance of other methods and properties. and we avoid overkill/change  `DocumentSet.add` and `ChunkSet.append`